app-backend/db/src/main/scala/AnnotationProjectDao.scala

```diff
@@ -89,55 +90,56 @@ object AnnotationProjectDao
       page: PageRequest,
       params: AnnotationProjectQueryParameters,
       user: User
-  ): ConnectionIO[PaginatedResponse[AnnotationProject.WithRelated]] =
-    authQuery(
+  ): ConnectionIO[
+    PaginatedResponse[AnnotationProject.WithRelatedAndSummary]
+  ] =
+    (authQuery(
       user,
       ObjectType.AnnotationProject,
       params.ownershipTypeParams.ownershipType,
       params.groupQueryParameters.groupType,
       params.groupQueryParameters.groupId
     ).filter(params)
-      .page(page)
-      .flatMap(toWithRelated)
-
-  def toWithRelated(
-      projectsPage: PaginatedResponse[AnnotationProject]
-  ): ConnectionIO[PaginatedResponse[AnnotationProject.WithRelated]] =
-    projectsPage.results.toList.toNel match {
-      case Some(projects) =>
-        projects traverse { project =>
-          for {
-            tileLayers <- TileLayerDao.listByProjectId(project.id)
-            labelClassGroups <- AnnotationLabelClassGroupDao.listByProjectId(
-              project.id
-            )
-            labelClassGroupsWithClasses <- labelClassGroups traverse {
-              labelClassGroup =>
-                AnnotationLabelClassDao.listAnnotationLabelClassByGroupId(
-                  labelClassGroup.id
-                ) map { cls =>
-                  labelClassGroup.withLabelClasses(cls)
+      .list(page.offset, page.limit, Page.createOrderClause(page))
+      .map { projectList =>
+        projectList traverse {
+          project =>
+            getWithRelatedAndSummaryById(project.id) map {
+              case Some(project) =>
+                val taskStatusList =
+                  params.projectFilterParams.taskStatusesInclude.toList
+                if (taskStatusList.size > 0) {
+                  taskStatusList.foldLeft(
+                    true
+                  )(
+                    (acc, status) =>
+                      (project.taskStatusSummary
+                        .getOrElse(status, 0) > 0) && acc
```

Comment:
If you reorder this (`acc && ...` instead of `(... && acc)`, you can skip evaluation of the task status map:

```scala
@ val nums = List(1, 2, 3, 4, 5)
nums: List[Int] = List(1, 2, 3, 4, 5)

@ def f(b: Boolean, n: Int) = {
    b && {
      println("lol")
      n < 3
    }
  }
defined function f

@ nums.foldLeft(true)(f)
lol
lol
lol
res4: Boolean = false

@ def g(b: Boolean, n: Int) = {
    {
      println("lol")
      n < 3
    } && b
  }
defined function g

@ nums.foldLeft(true)(g)
lol
lol
lol
lol
lol
res6: Boolean = false
```

The result is the same but the `&&` short circuits

Reply:
I'm not sure `&& acc` is correct here though -- what's the desired behavior? If I pass more than one task status, should I get only projects that have tasks in each of those task statuses? Or I should get projects with any tasks in any of the statuses I passed?

Reply:
_decided_: we don't know exactly what the future holds, so leave it `&&` for now until we figure out whether we definitely need `||`